src/client/datascience/jupyter/kernelVariables.ts

```diff
@@ -164,10 +164,6 @@ export class KernelVariables implements IJupyterVariables {
         // Import the data frame script directory if we haven't already
         await this.importDataFrameScripts(notebook);
 
-        if (targetVariable.rowCount) {
-            end = Math.min(end, targetVariable.rowCount);
-        }
```

Comment:
The row start and end however are already evaluated relative to the slice variable info, so this check is unnecessary.